tools/clipper.js

```diff
@@ -1,31 +1,123 @@
+	let doNotStitch = false;
+	let rescaleVideo = false;
+	if (extraArgs[0].value === 'Do Not Stitch Clips' || extraArgs[1].value === 'Do Not Stitch Clips') {
+		doNotStitch = true;
+	}
+	if (extraArgs[0].value === 'Rescale Video' || extraArgs[1].value === 'Rescale Video') {
+		rescaleVideo = true;
+	}
 
-	io.emit('request', {
-		internalId: `${id}_${nanoid()}`,
-		videoType,
-		videoLink,
-		timestamps,
-		fileName,
-		fileExt,
-	});
+	console.log(`Recieved Clipping Request ${internalId}, Is Multifile Clip: ${doNotStitch}, Is Rescaled: ${rescaleVideo}`);
+	if (await webdavClient.exists(`/TL Team/Projects/${projectName}/`) === false) {
+		// TODO: set up project directory with necessary documents via WebDAV
+		webdavClient.createDirectory(`/TL Team/Projects/${projectName}/`);
+	}
+	// This OS check is for development purposes only; will be removed in the future
+	if (os.platform() === 'win32') {
+		const cmd = `./clipper.ps1 -videotype ${videoType} -inlink ${videoLink} -timestampsIn "${timestamps}" -dlDir "../download/" -fulltitle ${internalId} -fileOutExt ${fileExt}`;
+		const res = exec(cmd, { shell: 'powershell.exe' }, async (error, stdout) => {
+			console.log(stdout);
+			console.log(error);
+			if (error !== null) {
+				console.log(error);
+				return 1;
+			}
+			if (doNotStitch === true) {
+				const zipFile = fs.createWriteStream('../download/clips.zip');
+				const archive = archiver('zip', {
+					zlib: { level: 9 },
+				});
+				archive.pipe(zipFile);
+				archive.directory('../download/', false);
+				archive.finalize();
+				const stream = fs.createReadStream('../download/clips.zip');
+				const result = await webdavClient.putFileContents('/TL Team/Projects/Test/clips.zip', stream);
+				if (result === false) {
+					return 1;
+				}
+				fs.unlink(`./download/${internalId}.${fileExt}`, (err) => {
+					if (err) console.log(err);
+				});
+			} else {
+				const stream = fs.createReadStream(`../download/${internalId}.${fileExt}`);
+				const result = await webdavClient.putFileContents(`/TL Team/Projects/Test/${projectName.replace(/\s+/g, '')}.${fileExt}`, stream);
+				if (result === false) {
+					return 1;
+				}
+				fs.unlink(`./download/${internalId}.${fileExt}`, (err) => {
+					if (err) console.log(err);
+				});
+			}
+			return 0;
+		});
+		if (res !== 0) return false;
+	} else {
+		const cmd = `pwsh ./clipper.ps1 -videotype ${videoType} -inlink ${videoLink} -timestampsIn "${timestamps}" -dlDir "../download/" -fulltitle ${internalId} -fileOutExt ${fileExt}`;
+		const res = exec(cmd, async (error, stdout) => {
+			console.log(stdout);
+			console.log(error);
+			if (error !== null) {
+				console.log(error);
+				return 1;
+			}
+			if (doNotStitch === true) {
+				const zipFile = fs.createWriteStream('../download/clips.zip');
+				const archive = archiver('zip', {
+					zlib: { level: 9 },
+				});
+				archive.pipe(zipFile);
+				archive.directory('../download/', false);
+				archive.finalize();
```

Comment:
https://www.archiverjs.com/docs/quickstart
Looking at API documentation, you should wait for the event, as the streams may not be finished yet after the finalize call.

Reply:
Implemented, but not sure if I did it correctly, so I will leave this unresolved until we can confirm it works.